src/generators/genlua.ml

```diff
@@ -1949,7 +1970,7 @@ let generate com =
         let file_content = Std.input_file ~bin:true path in
         print ctx "%s\n" file_content;
     in
-
+    print ctx "pcall(function () require\"luarocks.loader\" end)\n";
```

Comment:
You are missing the return of the require here. You don't need to wrap require un a function, the second argument of pcall can be arguments for the passed function.